server/src/routes/upload.ts

```diff
@@ -19,37 +19,49 @@ export function routes (app: any, session: any) {
   app.post('/sign-upload',
     session,
     async (req: express.Request, res: express.Response) => {
-      console.log(req.body)
       // Create a file object
-      const response = await adminMutate(
-        `${process.cwd()}/../gql/insertFile.gql`,
-        {
-          name: decodeURIComponent(req.body.filename),
-          uploaded_by_id: req.session.dbId,
-          asset_id: req.body.projectId
-        }
-      )
-
-      // Get the unique id of the upload object and make that the filename
-      const filename = response.returning[0].id
-
-      // Send signed url
-      s3Client.presignedPutObject(
-        'uploads', // Bucket name
-        encodeURIComponent(filename),
-        1000,
-        function (e, presignedUrl) {
-          if (e) return console.log(e)
-          res.json({
-            url: presignedUrl,
-            method: 'put',
-            fields: [],
-            headers: {
-              'content-type': req.body.contentType
+      try {
+        const bucketFound = await s3Client.bucketExists('uploads')
```

Comment:
This would avoid crashes if anything happens to Minio and the bucket is not accessible (Any transfer to nonexisting buckets will throw an exception) and it will allow us to throw error messages to the front end to temporarily deactivate uploads. An Ideal would be a try-catch block (But we are not handling errors yet).